app/routes/api/user.js

```diff
@@ -96,4 +98,34 @@ router.route('/users/:userId')
         });
 });
 
+router.route('/users/resetPassword')
+.put((request, response) => {
+    PasswordReset.findOne({resetToken: request.body.resetToken}).exec()
+    .then(function(passwordReset) {
+        User.findOne({email: passwordReset.email}).exec()
+        .then(function(user) {
+            user.setPassword(request.body.password, function(err, model, passwordErr) {
+                if (err || passwordErr) {
+                    response.status(400).end();
+                } else {
+                    model.update();
+                    passwordReset.remove();
+                    response.status(200).end();
+                }
+            });
+        });
+    });
+});
+
+router.route('/users/requestReset')
```

Comment:
Can we keep this with the /api/auth.js routes instead, since that's where the other password stuff is? Maybe at `/auth/password/requestReset`?